docs/framework/guides/custom-editor-creator.md

```diff
@@ -0,0 +1,429 @@
+
+```js
+import EditorUI from '@ckeditor/ckeditor5-core/src/editor/editorui';
+import enableToolbarKeyboardFocus from '@ckeditor/ckeditor5-ui/src/toolbar/enabletoolbarkeyboardfocus';
+import normalizeToolbarConfig from '@ckeditor/ckeditor5-ui/src/toolbar/normalizetoolbarconfig';
+import { enablePlaceholder } from '@ckeditor/ckeditor5-engine/src/view/placeholder';
+
+/**
+ * The multi-root editor UI class.
+ *
+ * @extends module:core/editor/editorui~EditorUI
+ */
+class MultirootEditorUI extends EditorUI {
+	/**
+	 * Creates an instance of the multi-root editor UI class.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor The editor instance.
+	 * @param {module:ui/editorui/editoruiview~EditorUIView} view The view of the UI.
+	 */
+	constructor( editor, view ) {
+		super( editor );
+
+		/**
+		 * The main (top–most) view of the editor UI.
+		 *
+		 * @readonly
+		 * @member {module:ui/editorui/editoruiview~EditorUIView} #view
+		 */
+		this.view = view;
+
+		/**
+		 * A normalized `config.toolbar` object.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		this._toolbarConfig = normalizeToolbarConfig( editor.config.get( 'toolbar' ) );
+	}
+
+	/**
+	 * Initializes the UI.
+	 */
+	init() {
+		const view = this.view;
+		const editor = this.editor;
+		const editingView = editor.editing.view;
+
+		view.render();
+
+		for ( const editable of this.view.editables ) {
+			// The editable UI element in DOM is available for sure only after the editor UI view has been rendered.
+			// But it can be available earlier if a DOM element has been passed to DecoupledEditor.create().
+			const editableElement = editable.element;
+
+			// Register the editable UI view in the editor. A single editor instance can aggregate multiple
+			// editable areas (roots) but the decoupled editor has only one.
+			this._editableElements.set( editable.name, editableElement );
+
+			// Let the global focus tracker know that the editable UI element is focusable and
+			// belongs to the editor. From now on, the focus tracker will sustain the editor focus
+			// as long as the editable is focused (e.g. the user is typing).
+			this.focusTracker.add( editableElement );
+
+			// Let the editable UI element respond to the changes in the global editor focus
+			// tracker. It has been added to the same tracker a few lines above but, in reality, there are
+			// many focusable areas in the editor, like balloons, toolbars or dropdowns and as long
+			// as they have focus, the editable should act like it is focused too (although technically
+			// it isn't), e.g. by setting the proper CSS class, visually announcing focus to the user.
+			// Doing otherwise will result in editable focus styles disappearing, once e.g. the
+			// toolbar gets focused.
+			editable.bind( 'isFocused' ).to( this.focusTracker );
```

Comment:
See my comment in `MultirootEditorUI#init()`. All editables cannot share the same `isFocused` state.